controllers/volumereplicationgroup_controller.go

```diff
@@ -887,6 +885,13 @@ func (v *VRGInstance) createOrUpdateVR(vrNamespacedName types.NamespacedName,
 			v.instance.Namespace, v.instance.Name, err)
 	}
 
+	if err := v.volRepStatusCheck(volRep); err != nil {
```

Comment:
(future) We will address this when we start updating status, possibly in the next PR

- We are checking the status of the just updated VR object, which will not have the latest status as per the API, so we need to fetch it and check if it agrees with our desired state
- We should check if state is as desired prior to returning from line 873 and not here, the intention would be we updated the object and as we own it, when its status changes we will be reconciled automatically, hence as of now we should assume VR has not reached desired state